src/org/zaproxy/zap/extension/ascanrulesAlpha/SQLInjectionMsSQL.java

```diff
@@ -0,0 +1,319 @@
+	static {
+		SQL_ERROR_TO_DBMS.put("com.microsoft.sqlserver.jdbc", "MsSQL");
+		SQL_ERROR_TO_DBMS.put("com.microsoft.jdbc", "MsSQL");
+		SQL_ERROR_TO_DBMS.put("com.inet.tds", "MsSQL");
+		SQL_ERROR_TO_DBMS.put("com.microsoft.sqlserver.jdbc", "MsSQL");
+		SQL_ERROR_TO_DBMS.put("com.ashna.jturbo", "MsSQL");
+		SQL_ERROR_TO_DBMS.put("weblogic.jdbc.mssqlserver", "MsSQL");
+		SQL_ERROR_TO_DBMS.put("[Microsoft]", "MsSQL");
+		SQL_ERROR_TO_DBMS.put("[SQLServer]", "MsSQL");
+		SQL_ERROR_TO_DBMS.put("[SQLServer 2000 Driver for JDBC]", "MsSQL");
+		SQL_ERROR_TO_DBMS.put("net.sourceforge.jtds.jdbc", "MsSQL");
+		SQL_ERROR_TO_DBMS.put("80040e14", "MsSQL");
+		SQL_ERROR_TO_DBMS.put("800a0bcd", "MsSQL");
+		SQL_ERROR_TO_DBMS.put("80040e57", "MsSQL");
+		SQL_ERROR_TO_DBMS.put("All queries in an SQL statement containing a UNION operator must have an equal number of expressions in their target lists", "MsSQL");
+		SQL_ERROR_TO_DBMS.put("All queries combined using a UNION, INTERSECT or EXCEPT operator must have an equal number of expressions in their target lists", "MsSQL");
+		//Note: only MSSQL mappings here.
+	}
+
+	/**
+	 * MsSQL specific time based injection strings.
+	 * Note: <<<<ORIGINALVALUE>>>> is replaced with the original parameter value at runtime in these examples below (see * comment)
+	 * TODO: Add SQL_MSSQL_TIME_REPLACEMENTS queries related to PARAM in SELECT/UPDATE/DELETE clause
+	*/
+	private static String[] SQL_MSSQL_TIME_REPLACEMENTS = {
+		ORIG_VALUE_TOKEN + " WAITFOR DELAY '" + SLEEP_TOKEN + "'" + SQL_ONE_LINE_COMMENT,						//Param in WHERE clause
+		ORIG_VALUE_TOKEN + "' WAITFOR DELAY '" + SLEEP_TOKEN + "'" + SQL_ONE_LINE_COMMENT,						//Param in WHERE clause
+		ORIG_VALUE_TOKEN + "\" WAITFOR DELAY '" + SLEEP_TOKEN + "'" + SQL_ONE_LINE_COMMENT,						//Param in WHERE clause
+		ORIG_VALUE_TOKEN + ") WAITFOR DELAY '" + SLEEP_TOKEN + "'" + SQL_ONE_LINE_COMMENT,
+		ORIG_VALUE_TOKEN + ") ' WAITFOR DELAY '" + SLEEP_TOKEN + "'" + SQL_ONE_LINE_COMMENT,
+		ORIG_VALUE_TOKEN + ") \" WAITFOR DELAY '" + SLEEP_TOKEN + "'" + SQL_ONE_LINE_COMMENT,
+		ORIG_VALUE_TOKEN + ")) WAITFOR DELAY '" + SLEEP_TOKEN + "'" + SQL_ONE_LINE_COMMENT,
+		ORIG_VALUE_TOKEN + ")) ' WAITFOR DELAY '" + SLEEP_TOKEN + "'" + SQL_ONE_LINE_COMMENT,
+		ORIG_VALUE_TOKEN + ")) \" WAITFOR DELAY '" + SLEEP_TOKEN + "'" + SQL_ONE_LINE_COMMENT,
+		ORIG_VALUE_TOKEN + ") WAITFOR DELAY '" + SLEEP_TOKEN + "' (" ,
+		ORIG_VALUE_TOKEN + ") ' WAITFOR DELAY '" + SLEEP_TOKEN + "' (",
+		ORIG_VALUE_TOKEN + ") \" WAITFOR DELAY '" + SLEEP_TOKEN + "' (",
+		ORIG_VALUE_TOKEN + ")) WAITFOR DELAY '" + SLEEP_TOKEN + "' ((",
+		ORIG_VALUE_TOKEN + ")) ' WAITFOR DELAY '" + SLEEP_TOKEN + "' ((",
+		ORIG_VALUE_TOKEN + ")) \" WAITFOR DELAY '" + SLEEP_TOKEN + "' ((",
+	};
+
+	/**
+	 * plugin dependencies (none! not even "SQL Injection")
+	 */
+	private static final String[] dependency = {};
+
+	/**
+	 * for logging.
+	 */
+	private static Logger log = Logger.getLogger(SQLInjectionMsSQL.class);
```

Comment:
Was referring to the variable name, but let it be as is (final should be added though).